incubator/elasticsearch/templates/client-deployment.yaml

```diff
@@ -131,6 +131,17 @@ spec:
           subPath: elasticsearch.keystore
           readOnly: true
 {{- end }}
+      - name: {{ .Values.prometheusExporter.clientName }}
```

Comment:
The metrics sidecar should be optional.
I suggest adding `Values.prometheusExporter.enabled` and using that to wrap the container in an if.

It should probably default to false.